app/src/main/java/io/homeassistant/companion/android/database/AppDatabase.kt

```diff
@@ -124,5 +128,32 @@ abstract class AppDatabase : RoomDatabase() {
                 }
             }
         }
+
+        private val MIGRATION_6_7 = object : Migration(6, 7) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                val cursor = database.query("SELECT * FROM sensors")
+                val sensors = mutableListOf<ContentValues>()
+                while (cursor.moveToNext()) {
+                    sensors.add(ContentValues().also {
+                        it.put("id", cursor.getString(cursor.getColumnIndex("unique_id")))
+                        it.put("enabled", cursor.getInt(cursor.getColumnIndex("enabled")))
+                        it.put("registered", cursor.getInt(cursor.getColumnIndex("registered")))
+                        it.put("state", "")
+                        it.put("state_type", "")
+                        it.put("type", "")
+                        it.put("icon", "")
+                        it.put("name", "")
+                        it.put("device_class", "")
+                    })
+                }
+                database.execSQL("DROP TABLE IF EXISTS `sensors`")
+                database.execSQL("CREATE TABLE IF NOT EXISTS `sensors` (`id` TEXT NOT NULL, `enabled` INTEGER NOT NULL, `registered` INTEGER NOT NULL, `state` TEXT NOT NULL, `state_type` TEXT NOT NULL, `type` TEXT NOT NULL, `icon` TEXT NOT NULL, `name` TEXT NOT NULL, `device_class` TEXT, `unit_of_measurement` TEXT, PRIMARY KEY(`id`))")
+                sensors.forEach {
+                    database.insert("sensors", OnConflictStrategy.REPLACE, it)
+                }
+
+                database.execSQL("CREATE TABLE IF NOT EXISTS `sensor_attributes` (`sensor_id` TEXT NOT NULL, `name` TEXT NOT NULL, `value` TEXT NOT NULL, PRIMARY KEY(`sensor_id`, `name`))")
```

Comment:
Shouldn't the cursor be closed or does it close automatically?

Reply:
Good catch, I thought I read somewhere you didn't need to close it but I'll close it anyway.